File: frontend/src/types/session.ts
export interface ColumnMap {
  time_col: string | null;
  front_raw_col: string;
  rear_raw_col: string;
  gyro_y_col: string;
  accel_x_col: string;
  accel_y_col: string;
  accel_z_col: string;
  invert_front: boolean;
  invert_rear: boolean;
}

export interface Session {
  id: string;
  name: string;
  bike_slug: string;
  csv_path: string;
  column_map: ColumnMap;
  velocity_quantity: 'wheel' | 'shaft';
  created_at: string;
  analyzed: boolean;
}

File: backend/app/models/session.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class ColumnMap(BaseModel):
    time_col: str | None = None
    front_raw_col: str = "front_raw"
    rear_raw_col: str = "rear_raw"
    gyro_y_col: str = "gyro_y_raw"
    accel_x_col: str = "accel_x_raw"
    accel_y_col: str = "accel_y_raw"
    accel_z_col: str = "accel_z_raw"
    invert_front: bool = False
    invert_rear: bool = False


class Session(BaseModel):
    id: str
    name: str
    bike_slug: str
    csv_path: str
    column_map: ColumnMap = Field(default_factory=ColumnMap)
    velocity_quantity: Literal["wheel", "shaft"] = "shaft"
    created_at: datetime
    analyzed: bool = False
